instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/trace/SpanConfigE2ETests.kt

```diff
@@ -0,0 +1,239 @@
+    fun trace_config_set_bundle_with_rum_enabled() {
+        // this one will have application_id tag
+        val testMethodName = "trace_config_set_bundle_with_rum_enabled"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                tracerProvider = { AndroidTracer.Builder().setBundleWithRumEnabled(true).build() }
+            )
+        }
+        GlobalTracer.get()
+            .buildSpan(testMethodName)
+            .start()
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: AndroidTracer#Builder#fun setBundleWithRumEnabled(Boolean): Builder
+     */
+    @Test
+    fun trace_config_set_bundle_with_rum_disabled() {
+        // this one won't have application_id tag
+        val testMethodName = "trace_config_set_bundle_with_rum_disabled"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                tracerProvider = { AndroidTracer.Builder().setBundleWithRumEnabled(false).build() }
+            )
+        }
+        GlobalTracer.get()
+            .buildSpan(testMethodName)
+            .start()
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: AndroidTracer#Builder#fun addGlobalTag(String, String): Builder
+     */
+    @Test
+    fun trace_config_add_global_tag() {
+        val testMethodName = "trace_config_add_global_tag"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                tracerProvider = {
+                    AndroidTracer.Builder()
+                        .addGlobalTag(
+                            SPECIAL_STRING_TAG_NAME,
+                            "str${forge.anAlphaNumericalString()}"
+                        )
+                        .build()
+                }
+            )
+        }
+        GlobalTracer.get()
+            .buildSpan(testMethodName)
+            .start()
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: AndroidTracer#Builder#fun setServiceName(String): Builder
+     */
+    @Test
+    fun trace_config_set_service_name() {
+        val testMethodName = "trace_config_set_service_name"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                tracerProvider = {
+                    AndroidTracer.Builder()
+                        .setServiceName("service-$testMethodName")
+                        .build()
+                }
+            )
+        }
+        GlobalTracer.get()
+            .buildSpan(testMethodName)
+            .start()
+            .finish()
+    }
+
+    /**
+     * apiMethodSignature: Datadog#fun setUserInfo(String? = null, String? = null, String? = null, Map<String, Any?> = emptyMap())
+     */
+    @Test
+    fun trace_config_set_user_info() {
```

Comment:
Ah...good catch, I think I forgot to add this `test-case` for Logs. I will add it in my current PR.